### DISCUSSION TITLE
Programmatic zoom/pan

### DISCUSSION THREAD

**Discussion by cornhundred** [visgl/deck.gl#5926] [Q&A] [answered]
Hi, how can I perform a programmatic zoom in deck.gl (I also want to allow user controlled zooming). I wou like to have a visual start with a programmatic zoom (using an easing function) that helps the user see the scope of the visual, but also allow them to zoom/pan into the visual.

**Comment by jfuehner** [comment, marked as answer]
@cornhundred you can simply update the `viewState` object with a new zoom level…

[https://deck.gl/docs/developer-guide/views](https://deck.gl/docs/developer-guide/views) 

```
deck.setProps({
        viewState: {zoom}
      });
```

**Reply by cornhundred** [reply]
thanks, I'll try that out

**Comment by Pessimistress** [comment]
https://deck.gl/docs/developer-guide/interactivity#controlling-the-camera
